runtime.md

```diff
@@ -34,22 +34,24 @@ This is provided so that consumers can find the container's configuration and ro
 ```
 
 ## Lifecycle
-
-### Create
-
-Creates the container: file system, namespaces, cgroups, capabilities.
-
-### Start (process)
-
-Runs a process in a container.
-Can be invoked several times.
-
-### Stop (process)
-
-Not sure we need that from runc cli.
-Process is killed from the outside.
-
-This event needs to be captured by runc to run onstop event handlers.
+The lifecycle describes the timeline of events that happen from when a container is created to when it ceases to exist.
+
+1. OCI compliant runtime is invoked by passing the bundle path as argument.
+2. The container's runtime environment is created according to the configuration in config.json.
+   Any updates to config.json after container is running do not affect the container.
+3. The container's state.json file is written to the filesystem under /run/opencontainer/<runtime>/containers/<id>/.
+4. The prestart hooks are invoked by the runtime.
+   If any prestart hook fails, then the container is killed and the lifecycle continues at step 8.
+5. The user specified process is executed in the container.
+6. The poststart hooks are invoked by the runtime.
+   If any poststart hook fails, then the container is killed and the lifecycle continues at step 8.
+7. Additional actions such as pausing the container, resuming the container or signaling the container may be performed using the runtime interface.
+   The container could also error out or crash.
+8. The container is destroyed by undoing the steps performed during create phase.
```

Comment:
On Wed, Dec 02, 2015 at 10:48:14AM -0800, Daniel, Dao Quang Minh wrote:
> > +2. The container's runtime environment is created according to the configuration in config.json.
> > +   Any updates to config.json after container is running do not affect the container.
>
> create phase are not too clear to me i.e what are the steps in the
> create phase. Seems like 2->6 but it's nicer to be sure.

I'd assumed it was just step 2, since that's the only one that
mentions “create” except for step 8's reference to undoing the
creation.  But yeah, adding explicit step numbers to any step
references seems like an easy way to clarify.